components/self_hosted_purchase_modal/seats_calculator.tsx

```diff
@@ -109,10 +128,10 @@ export default function SeatsCalculator(props: Props) {
             // nulls out the customMessage. By forcefully creating a new react element error,
             // it will trigger the error still existing, and the error will keep being shown
             // in the input component
-            props.onChange(validateSeats(props.seats.quantity, annualPricePerSeat, props.existingUsers));
+            props.onChange(validateSeats(props.seats.quantity, annualPricePerSeat, props.existingUsers, props.isCloud));
             return;
         }
-        props.onChange(validateSeats(value, annualPricePerSeat, props.existingUsers));
+        props.onChange(validateSeats(value, annualPricePerSeat, props.existingUsers, props.isCloud));
     };
 
     const maxSeats = calculateMaxUsers(annualPricePerSeat);
```

Comment:
Looks like the total calculation on the next line will need to use `parseFloat` now to support the possibility of decimals. Otherwise, values like `7.5` get truncated to `7`, and the calculator gets off.